app/src/main/java/org/simlar/service/liblinphone/LinphoneHandler.java

```diff
@@ -377,4 +383,150 @@ private static void enableDebugMode(final boolean enabled)
 	{
 		LinphoneCoreFactory.instance().setDebugMode(enabled, "DEBUG");
 	}
+
+	public void requestVideoUpdate(final boolean enable)
+	{
+		Lg.i("requestVideoUpdate: enable=", enable);
+
+		final LinphoneCall currentCall = getCurrentCall();
+		if (currentCall == null) {
+			Lg.w("no current call to add video to");
+			return;
+		}
+
+		final LinphoneCallParams params = currentCall.getCurrentParams();
+		if (params.getVideoEnabled() == enable) {
+			Lg.i("requestVideoUpdate already: ", enable, " => aborting");
+			return;
+		}
+
+		params.setVideoEnabled(enable);
+		mLinphoneCore.updateCall(currentCall, params);
+	}
+
+	public void preventAutoAnswer()
+	{
+		Lg.i("preventAutoAnswer");
+
+		final LinphoneCall currentCall = getCurrentCall();
+		if (currentCall == null) {
+			Lg.w("no current call to prevent auto answer for");
+			return;
+		}
+
+		try {
+			mLinphoneCore.deferCallUpdate(currentCall);
+		} catch (final LinphoneCoreException e) {
+			Lg.ex(e, "LinphoneCoreException during deferCallUpdate");
+		}
+	}
+
+	public void acceptVideoUpdate(final boolean accept)
+	{
+		Lg.i("acceptVideoUpdate accept=", accept);
+
+		final LinphoneCall currentCall = getCurrentCall();
+		if (currentCall == null) {
+			Lg.w("no current call to accept video for");
+			return;
+		}
+
+		final LinphoneCallParams params = currentCall.getCurrentParams();
+		if (accept) {
+			params.setVideoEnabled(true);
+		}
+
+		try {
+			mLinphoneCore.acceptCallUpdate(currentCall, params);
+		} catch (final LinphoneCoreException e) {
+			Lg.ex(e, "LinphoneCoreException during acceptCallUpdate");
+		}
+	}
+
+	public void setVideoWindow(final Object videoWindow)
+	{
+		Lg.i("setVideoWindow");
+
+		if (mLinphoneCore == null) {
+			Lg.e("setVideoWindow: mLinphoneCore is null => aborting");
+			return;
+		}
+
+		mLinphoneCore.setVideoWindow(videoWindow);
+	}
+
+	public void setVideoPreviewWindow(final Object videoPreviewWindow)
+	{
+		Lg.i("setVideoPreviewWindow");
+
+		if (mLinphoneCore == null) {
+			Lg.e("setVideoPreviewWindow: mLinphoneCore is null => aborting");
+			return;
+		}
+
+		enableCamera(videoPreviewWindow != null);
+		mLinphoneCore.setPreviewWindow(videoPreviewWindow);
+	}
+
+	private void setFrontCameraAsDefault()
+	{
+		int cameraId = 0;
+		for (final AndroidCameraConfiguration.AndroidCamera androidCamera : AndroidCameraConfiguration.retrieveCameras()) {
+			if (androidCamera.frontFacing) {
+				cameraId = androidCamera.id;
```

Comment:
Is it correct that to choose choose the **last** found front facing camera?

Reply:
Honestly, I do not know :)
Linphone-Android does it that way. But I think, taking the first front facing camera is a good idea.

I have pushed a commit to the devel-video branch:
7780aa67286a5adf87096665e6b1666ec9be037d